test/tests/functional/pbs_job_array.py

```diff
@@ -563,3 +563,42 @@ def test_qmove_job_array(self):
         self.server.manager(MGR_CMD_SET, SERVER, a)
         self.server.expect(JOB, {'job_state': 'S'}, subjid_1)
         self.server.expect(JOB, {'job_state': 'R'}, subjid_3)
+
+    def test_delete_history_subjob_server_restart(self):
+        """
+        Test that subjobs can be deleted from history
+        and they remain delted after server restart
+        """
+        a = {'job_history_enable': 'true'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, self.mom.shortname)
+        j = Job(TEST_USER, attrs={
+            ATTR_J: '1-2', 'Resource_List.select': 'ncpus=1'})
+        j.set_sleep_time(5)
+        j_id = self.server.submit(j)
+        self.server.expect(JOB, {'job_state': 'F'}, j_id, extend='x', offset=5)
+        self.server.delete(j_id, extend='deletehist')
+        self.server.expect(JOB, 'job_state', op=UNSET, id=j_id)
```

Comment:
I think you need to say extend=x.  Even if it wasn't deleted from the history, this will still pass because the job won't show up if you don't show history.